client/src/views/BlockDetail.vue

```diff
@@ -142,48 +153,63 @@ export default {
       blockInfo: {},
       tableData: {
         head: [
-          '#',
-          'Timestamp',
-          'Transaction Type',
-          'Fees',
-          'Signer',
-          'Transaction Hash'
+          "#",
+          "Timestamp",
+          "Transaction Type",
+          "Fees",
+          "Signer",
+          "Transaction Hash"
         ],
         data: []
       },
       loading: true
-    }
+    };
+  },
+  watch: {
+    $route: "asyncData"
   },
   mounted() {
-    this.asyncData()
+    this.asyncData();
   },
   methods: {
+    previousBlcok() {
+      var previousBlcok = Number(this.blockId) - 1;
+      this.$router.push({
+        path: "/block/" + previousBlcok
+      });
+    },
+    nextBlock() {
+      var nextBlock = Number(this.blockId) + 1;
+      this.$router.push({
+        path: "/block/" + nextBlock
+      });
+    },
     asyncData() {
-      let self = this
-      this.blockId = this.$route.params.blockId
-      this.loading = 0
-      DataService.getBlockInfo(this.blockId).then(function (data) {
-        self.blockInfo = data.blockInfo
+      let self = this;
+      this.blockId = this.$route.params.blockId;
```

Comment:
if self = this, why using this.blockId and not self.blockId?

Reply:
we can't access the vue data inside the function by calling 'this' . Inside the " DataService.getBlockInfo(this.blockId)"  this referred to the function only. 
So we refer **this** to **self** to access inside the function.

Reply:
I see, thanks!